contracts/Everlong.sol

```diff
@@ -387,25 +407,108 @@ contract Everlong is IEverlong {
             if (!IHyperdrive(hyperdrive).isMature(position)) {
                 return output;
             }
-            output += IHyperdrive(hyperdrive).closeLong(asBase, position, "");
+            output += IHyperdrive(hyperdrive).closeLong(
+                asBase,
+                position,
+                0,
+                ""
+            );
             _portfolio.handleClosePosition();
         }
     }
 
     /// @dev Close positions until the targeted amount of output is received.
-    /// @param _targetOutput Minimum amount of proceeds to receive.
+    /// @param _targetOutput Target amount of proceeds to receive.
     /// @return output Total output received from closed positions.
     function _closePositions(
         uint256 _targetOutput
     ) internal returns (uint256 output) {
+        // Round `_targetOutput` up to Hyperdrive's minimum transaction amount.
+        _targetOutput = _targetOutput.max(
+            IHyperdrive(hyperdrive).getPoolConfig().minimumTransactionAmount
+        );
+
+        // Since multiple position's worth of bonds may need to be closed,
+        // iterate through each position starting with the most mature.
+        //
+        // For each position, use the expected output of closing the entire
+        // position to estimate the amount of bonds to sell for a partial closure.
+        IEverlong.Position memory position;
+        uint256 totalPositionValue;
+        uint256 expectedValuePerBond;
+        uint256 bondsNeeded;
         while (!_portfolio.isEmpty() && output < _targetOutput) {
-            output += IHyperdrive(hyperdrive).closeLong(
+            // Retrieve the most mature position.
+            position = _portfolio.head();
+
+            // Estimate the value of the entire position, and use it to derive
+            // the expected output per bond.
+            totalPositionValue = IHyperdrive(hyperdrive).previewCloseLong(
                 asBase,
-                _portfolio.head(),
+                position,
                 ""
             );
-            _portfolio.handleClosePosition();
+            expectedValuePerBond = totalPositionValue.mulDivDown(
```

Comment:
Do we just need this for rounding issues, or is there another reason? Naively, I'd think that the value per bond will always be greater than or equal to expected if all of our math used infinite precision.